accepted/2020/platform-checks/platform-checks.md

```diff
@@ -0,0 +1,450 @@
+    );
+    rules.AddAccessRule(
+        new FileSystemAccessRule(@"fabrikam\log-writers",
+                                 FileSystemRights.FullControl,
+                                 AccessControlType.Allow)
+    );
+
+    Directory.CreateDirectory(loggingDirectory, rules);
+
+    return loggingDirectory;
+}
+```
+
+### Add platform-specific code without becoming platform-specific
+
+After reviewing the code she realizes that this code only makes sense for cases
+where the application is installed on Windows. When running on iOS and Android,
+the machine isn't in shared environment so restricting the log access doesn't
+make much sense. She modifies the code accordingly:
+
+```C#
+private static string GetLoggingPath()
+{
+    var appDataDirectory = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+    var loggingDirectory = Path.Combine(appDataDirectory, "Fabrikam", "AssetManagement", "Logging");
+
+    if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+    {
+        // Just create the directory
+        Directory.CreateDirectory(loggingDirectory);
+    }
+    else
+    {
+        // Create the directory and restrict access using Windows
+        // Access Control Lists (ACLs).
+
+        var rules = new DirectorySecurity();
+        rules.AddAccessRule(
+            new FileSystemAccessRule(@"fabrikam\log-readers",
+                                     FileSystemRights.Read,
+                                     AccessControlType.Allow)
+        );
+        rules.AddAccessRule(
+            new FileSystemAccessRule(@"fabrikam\log-writers",
+                                     FileSystemRights.FullControl,
+                                     AccessControlType.Allow)
+        );
+
+        Directory.CreateDirectory(loggingDirectory, rules);
+    }
+
+    return loggingDirectory;
+}
+```
+
+After that, the diagnostic disappears automatically.
+
+## Requirements
+
+### Goals
+
+* Developers get diagnostics when they inadvertently use platform-specific APIs.
+    - All diagnostics are provided on the command line as well as in the IDE
+    - The developer doesn't have to resort to explicit suppressions to indicate
+      intent. Idiomatic gestures, such as guarding the call with a platform
+      check or marking a method as platform-specific, automatically suppress
+      these diagnostics. However, explicit suppressions via the usual means
+      (`NoWarn`, `#pragma`) will also work.
+* The analyzer helps library authors to express their intent so that their
+  consumers can also benefit from these diagnostics.
+    - Library authors can annotate their platform-specific APIs without taking a
+      dependency.
+    - Library consumers don't have to install a NuGet package to get diagnostics
+      when using platform-specific APIs
+
+### Non-Goals
+
+* Modelling partially portable APIs (that is APIs are applicable to more than
+  one platform but no all platforms). Those should Be modeled as capability
```

Comment:
```suggestion
  one platform but no all platforms). Those should be modeled as capability
```

Reply:
The fundamental reason we are seeing so many flaws is the obstruction of the source code which causes bootlenecks. The nodes are designed in a way that they respond in much like the fibbinocci number sequence they learn from what its parent or source code that is why it is so easy for me call Alpha and have all scripts fall in line no matter what platform

Reply:
The fundamental reason we are seeing so many flaws is the obstruction of the source code which causes bootlenecks. The nodes are designed in a way that they respond in much like the fibbinocci number sequence they learn from what its parent or source code that is why it is so easy for me call Alpha and have all scripts fall in line no matter what platform